### PR TITLE
l-03: add _isExpiredDueToGracePeriod to check expiry of queued rollback

### PR DESCRIPTION
**Pull request by thelostone-mc**

### CODE REVIEW

File: test/RollbackManagerTimelockCompound.unit.t.sol [hunk 1]
```diff
@@ -139,11 +139,201 @@ contract Execute is ExecuteBase, RollbackManagerTimelockCompoundTest {
...
+
+    // Should be in Queued state (within grace period)
+    assertEq(uint8(rollbackManager.state(rollbackId)), uint8(IGovernor.ProposalState.Queued));
+
+    // Execute should succeed
+    vm.prank(guardian);
+    rollbackManager.execute(_targets, _values, _calldatas, _description);
+  }
+
+  function testFuzz_ExecuteRollbackAfterGracePeriod(
```
Comment (apbendi) [inline test/RollbackManagerTimelockCompound.unit.t.sol:176]: nit: since we're expecting a revert, this test should have a `RevertIf`  or `RevertWhen` prefix, maybe something like `testFuzz_RevertWhen_TheGracePeriodHasElapsed`

File: test/RollbackManagerTimelockCompound.unit.t.sol [hunk 2]
```diff
@@ -139,11 +139,201 @@ contract Execute is ExecuteBase, RollbackManagerTimelockCompoundTest {
...
+    // Should be in Expired state (past grace period)
+    assertEq(uint8(rollbackManager.state(rollbackId)), uint8(IGovernor.ProposalState.Expired));
+
+    // Execute should revert because state is not Queued
+    vm.prank(guardian);
+    vm.expectRevert(abi.encodeWithSelector(RollbackManager.RollbackManager__NotQueued.selector, rollbackId));
+    rollbackManager.execute(_targets, _values, _calldatas, _description);
+  }
+
+  function testFuzz_GracePeriodExpiryBoundary(
```
Comment (apbendi) [inline test/RollbackManagerTimelockCompound.unit.t.sol:210]: The method under test here seems to be the `state` method, not the `execute` method, yet this is in the `Execute` test contract

Comment (apbendi) [reply test/RollbackManagerTimelockCompound.unit.t.sol:210]: Also, why test the boundary explicitly. If we have a test for before, and one for after, make sure the exact GracePeriod test is included in the bound on whichever side we intend the functionality to operate.

File: test/RollbackManagerTimelockControl.unit.t.sol [hunk 3]
```diff
@@ -157,6 +157,65 @@ contract Execute is ExecuteBase, RollbackManagerTimelockControlTest {
       assertEq(_lastExecuteBatchCall.calldatas[i], _calldatas[i]);
     }
   }
+
+  function testFuzz_ExecuteRollbackWithinGracePeriod(
```
Comment (apbendi) [inline test/RollbackManagerTimelockControl.unit.t.sol:161]: nit: naming here could be better

File: test/RollbackManagerTimelockControl.unit.t.sol [hunk 4]
```diff
@@ -157,6 +157,65 @@ contract Execute is ExecuteBase, RollbackManagerTimelockControlTest {
...
+
+    // Should be in Queued state (no upper bound for OZ)
+    assertEq(uint8(rollbackManager.state(rollbackId)), uint8(IGovernor.ProposalState.Queued));
+
+    // Execute should succeed
+    vm.prank(guardian);
+    rollbackManager.execute(_targets, _values, _calldatas, _description);
+  }
+
+  function testFuzz_NoGracePeriodExpiry(
```
Comment (apbendi) [inline test/RollbackManagerTimelockControl.unit.t.sol:193]: nit: naming
